src/vs/platform/windows/electron-main/window.ts

```diff
@@ -441,16 +443,45 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 			return false;
 		};
 
-		const isRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
+		const isSvgRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
 			return details.resourceType === 'xhr' || isSafeFrame(details.frame);
 		};
 
+		const workbenchRootUrl = FileAccess.asBrowserUri('vs/', require).toString(true);
+		const isAllowedVsCodeFileRequest = (uri: URI, details: Electron.OnBeforeRequestListenerDetails) => {
+			// Allow vscode-file requests from main window
+			if (details.frame?.url.startsWith('vscode-file://vscode-app/')) {
+				return true;
+			}
+
+			if (details.frame?.url || details.frame?.parent) { // Coming from unrecognized frame
+				return false;
+			}
+
+			// Dealing with frame without a url set yet. This happens when first loading a window.
+			// Here we still want to allow loading the main workbench html files
+			if (details.url.startsWith(workbenchRootUrl) && uri.path.endsWith('.html')) {
```

Comment:
Do we understand this properly?

Reply:
I am also unsure which case does this represent ? Can you give an example of which frame is this referring to ?

Reply:
This is to handle the request to load the initial html contents of a window (typically this is `workbench.html`, but it can also be something like `processExplorer.html` )

Without this, this initial request ends up being blocked since `details.frame.url` is empty